src/platform/nrfconnect/BUILD.gn

```diff
@@ -47,7 +47,7 @@ static_library("nrfconnect") {
     "SystemPlatformConfig.h",
   ]
 
-  deps = []
+  deps = [ "${chip_root}/src/setup_payload" ]
```

Comment:
Should we somehow separate out setup_payload? right now it contains very high level code, including code that was explicitly claimed to only ever run on "big devices" (uses std::string).

It seems desirable to have one portion with data types but small device safe and the other portion with the bigger  changes. Having a platform depend on the entire setup payload seems not ideal - thinking that other platforms may attempt to follow suite and some may have less resources than nrf boards.